src/main/java/net/wimpi/modbus/io/ModbusTCPTransaction.java

```diff
@@ -187,7 +187,9 @@ public void execute() throws ModbusIOException, ModbusSlaveException, ModbusExce
                     m_Connection.connect();
                     m_IO = m_Connection.getModbusTransport();
                 } catch (Exception ex) {
-                    throw new ModbusIOException("Connecting failed.");
+                    final String errMsg = "Connecting failed.";
```

Comment:
In other similar places the cause message was appended to the error message of rethrown exception. This could be useful also in here.

--

EDIT: Actually, there are also many more where it was not appended. So this is rather inconsistent.

Reply:
I'd rather keep our changes to minimum. There are inconsistencies all over this library.

Reply:
Alright